src/interfaces.d.ts

```diff
@@ -35,18 +44,18 @@ export interface HasSortEvent {
 }
 
 export interface HasItem {
-	item: ItemProperties<any>;
+	item: ItemProperties;
 }
 
 export interface HasItems {
-	items: ItemProperties<any>[];
+	items: ItemProperties[];
 }
 
 export interface HasValue {
 	value: string;
 }
 
-export interface ItemProperties<T> {
+export interface ItemProperties<T = any> {
```

Comment:
Am I understanding correctly that dgrid requires data objects to look like this:
```javascript
{
    id: <idValue>,
    data: {
        <... data properties>
    }
}
```

Are we not able to continue dgrid 1's design of accepting flatter objects (with support for `idProperty` as well):

```javascript
{
    id: <idValue>,
    <... data properties>
}
```